schema/descriptor.schema.json

```diff
@@ -28,18 +28,57 @@
             "description": "A string that describes the tool command line, where input and output values are identified by \"keys\". At runtime, command-line keys are substituted with flags and values.",
             "type": "string"
         },
-        "docker-image": {
-            "id": "http://github.com/boutiques/boutiques-schema/docker-image",
-            "minLength": 1,
-            "description": "Name of a Docker image where tool is installed and configured. Example: docker.io/neurodebian.",
-            "type": "string"
-        },
-        "docker-index": {
-            "id": "http://github.com/boutiques/boutiques-schema/docker-index",
-            "minLength": 1,
-            "description": "Docker index where Docker image is available.",
-            "default": "http://index.docker.io",
-            "type": "string"
+        "container": {
+            "id": "http://github.com/boutiques/boutiques-schema/container",
+            "type": "object",
+            "description": "Object representing a container for the tool.",
+            "allOf": [{
```

Comment:
It allows potentially letting `container-hash` and `launch-directory` coexist with either the `docker` spec or the `rootfs/singularity` spec. It does not force them to be required. It just lets both possible specs match to these so that the same values dont have to be written out twice (i.e. once in each of the two container type specs).